CMakeLists.txt

```diff
@@ -71,47 +83,63 @@ if (BUILD_TESTING)
   add_test(edlib_tests ${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/runTests)
 endif()
 
-if (NOT WIN32) # If on windows, do not build binaries that do not support windows.
-  add_executable(edlib-aligner apps/aligner/aligner.cpp)
-  target_link_libraries(edlib-aligner edlib)
+if(EDLIB_BUILD_UTILITIES)
+  if(NOT WIN32) # If on windows, do not build binaries that do not support windows.
+    add_executable(edlib-aligner apps/aligner/aligner.cpp)
+    target_link_libraries(edlib-aligner edlib)
+  endif()
 endif()
 
 # configure and install pkg-config file
+set(PKG_EDLIB_DEFS "")
+get_target_property(EDLIB_INTERFACE_DEFS edlib INTERFACE_COMPILE_DEFINITIONS)
+if(NOT (EDLIB_INTERFACE_DEFS STREQUAL "NOTFOUND" OR EDLIB_INTERFACE_DEFS STREQUAL "EDLIB_INTERFACE_DEFS-NOTFOUND"))
+  foreach(EDLIB_INTERFACE_DEF ${EDLIB_INTERFACE_DEFS})
+    list(APPEND PKG_EDLIB_DEFS "-D${EDLIB_INTERFACE_DEF}")
+  endforeach()
+endif()
+string(REPLACE ";" " " PKG_EDLIB_DEFS "${PKG_EDLIB_DEFS}")
```

Comment:
remind me again why we need this?

Reply:
this block extracts interface definitions of `edlib` target and formats them as `-D<def1> -D<def2> ...` in `PKG_EDLIB_DEFS` so that it can be injected in pkg-config file through `configure_file` (variable substitution.

Reply:
Would it be a problem if `EDLIB_API == __declspec(dllimport)` when static linking? I've never really used MSVC, and don't know exactly when it complains on `__declspec(dllimport)` decorators.

Reply:
No, when using `__declspec(dllimoprt)`, the compiler+linker will look for a symbol `_imp_symbol`.
This causes a missing symbol link error.
Google for `LNK2019 _imp_ static` to get an idea what kind of errors we're trying to avoid.

[1280 hits on stackoverflow](https://www.google.com/search?q=lnk2019+_imp+static+site%3Astackoverflow.com)! Wow.